### PR TITLE
Add E0463 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth 

Also: should I merged both commits? Not sure if it's really useful to keep the first one.

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -2233,9 +2233,9 @@ register_diagnostics! {
     E0285, // overflow evaluation builtin bounds
     E0298, // mismatched types between arms
     E0299, // mismatched types between arms
-    E0300, // unexpanded macro
-    E0304, // expected signed integer constant
-    E0305, // expected constant
+    // E0300, // unexpanded macro
```
Comment (Manishearth) [inline src/librustc/diagnostics.rs]: why comment these out?

File: src/librustc_metadata/diagnostics.rs [hunk 2]
```diff
@@ -56,6 +56,18 @@ you want. Example:
 ```
 "##,
 
+E0463: r##"
+A plugin was called but cannot be found. Erroneous code example:
```
Comment (Manishearth) [inline src/librustc_metadata/diagnostics.rs]: Just mention crate here

File: src/librustc_metadata/diagnostics.rs [hunk 3]
```diff
@@ -56,6 +56,18 @@ you want. Example:
...
+
+```
+#![feature(plugin)]
```
Comment (Manishearth) [inline src/librustc_metadata/diagnostics.rs]: Give examples for plugins and crates

File: src/librustc_metadata/diagnostics.rs [hunk 4]
```diff
@@ -56,6 +56,18 @@ you want. Example:
...
+#![plugin(cookie_monster)] // error: can't find crate for `cookie_monster`
+```
+
+In Rust, plugins are handled the same way as crate. You need to link your
+code to it in order to be able to use it (through Cargo for example).
```
Comment (Manishearth) [inline src/librustc_metadata/diagnostics.rs]: Mention the `-L` flag too

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by Manishearth** [comment]
This still doesn't talk about crates, and focuses on plugins.

If the error really only occurs for plugins, the inline error should be improved first. If not, we should mention crates as well.

File: src/librustc_metadata/diagnostics.rs [hunk 5]
```diff
@@ -56,6 +56,19 @@ you want. Example:
 ```
 "##,
 
+E0463: r##"
+A plugin/crate was called but cannot be found. Erroneous code example:
```
Comment (Manishearth) [inline src/librustc_metadata/diagnostics.rs]: was declared but not found

File: src/librustc_metadata/diagnostics.rs [hunk 6]
```diff
@@ -56,6 +56,19 @@ you want. Example:
...
+
+```
+#![feature(plugin)]
+#![plugin(cookie_monster)] // error: can't find crate for `cookie_monster`
+```
+
+In Rust, plugins are handled the same way as crate. You need to link your
```
Comment (Manishearth) [inline src/librustc_metadata/diagnostics.rs]: You need to link your code to the relevant crate in order to be able to use it (through Cargo or ....). Plugins are crates as well, and you link to them the same way.

**Pull request merged**